### PR TITLE
Persist wizard

### LINKED ISSUE #131: Users don't expect 'back' button in browser to immediately exit the wizard
If I'm in the middle of the wizard, the back button will immediately take me to the previous page, and my progress through the wizard will be lost. A better behavior would be to take me one step back in the wizard.

### PR DESCRIPTION
**Pull request by AmitJoki**
Fixes #131

### CODE REVIEW

File: app/assets/javascripts/components/wizard/wizard.jsx [hunk 1]
```diff
@@ -19,6 +19,24 @@ const getState = () =>
...
+  window.onpopstate = function (event) { // Listen to changes
+    if (event.state) {
+      WizardActions.goToWizard(event.state.index);
+    }
+  };
+};
+
+const unloadEvents = function () {
+    window.onpopstate = null;
+    window.onbeforeunload = null;
```
Comment (ragesoss) [inline app/assets/javascripts/components/wizard/wizard.jsx]: These should probably be camelCase?

Comment (AmitJoki) [reply app/assets/javascripts/components/wizard/wizard.jsx]: @ragesoss They are vanilla JS events. They're not something which a user can define. We can only listen to these events. See [onpopstate](https://developer.mozilla.org/en-US/docs/Web/API/Window/onpopstate) & [onbeforeunload](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload)

**Comment by ragesoss** [comment]
This works really nicely! The only glitch I notice is that the step in the page title does not change when you use forward/back. If you can consolidate that into the main functions for changing steps, that would be good. (If not, the title steps could be removed.)

**Comment by AmitJoki** [comment]
@ragesoss can you please check it now? Hopefully, the titles must now be synced. If you still find some errors, we could probably take the title steps off.

**Comment by ragesoss** [comment]
@AmitJoki it works for most parts of the wizard, but it gets out of sync when you navigate to an earlier step from the final Summary step. The step number rendered in the panel itself is always correct, though. Is there a good way to tie it to that?

**Comment by AmitJoki** [comment]
@ragesoss everything is fixed now. I ran it locally and every step is getting synced.

**Pull request merged**